app/controllers/job_runs_controller.rb

```diff
@@ -48,4 +49,10 @@ def download_report
   def job_run_params
     params.require(:job_run).permit(:batch_context_id, :job_type)
   end
+
+  def structural_changed?
+    return false unless @discovery_report
+
+    !@discovery_report['rows'].each.with_object('file_diffs').map(&:[]).compact.empty?
```

Comment:
Can you say more about what all this line is doing? Alternatively, a test covering this method might do the trick, if you're at the point of writing tests for this PR in your workflow.

Reply:
I should write a test for this. What this is doing is determining if there are ANY structural changes in the JSON - it's somewhat what complex as the changes are in each individual druid element.

Reply:
Word. A test showing the diff data shapes and how they affect the return value of this method would be perfect---definitely superior to us jabbering about code through GitHub.